src/view/gui/fMainWindow.cpp

```diff
@@ -4462,24 +4460,113 @@ void fMainWindow::PCAEstimateOnExistingModel(const std::string &modeldirectory,
 }
 
 
-void fMainWindow::CallGeneratePopualtionAtlas(const std::string inputdirectory, const std::string inputlabel, const std::string inputatlas, const std::string outputdirectory)
+void fMainWindow::CallGeneratePopualtionAtlas(const std::string inputFileName, const std::string inputatlas, const std::string outputdirectory)
 {
-  if (!cbica::isDir(inputdirectory))
+  if (!cbica::isFile(inputFileName))
   {
-    ShowErrorMessage("Input directory passed is not a valid directory, please re-check", this);
+    ShowErrorMessage("Input Batch file passed is not a valid file, please re-check", this);
     return;
   }
-  if (!cbica::isFile(inputlabel))
+  if (!cbica::isFile(inputatlas))
   {
-    ShowErrorMessage("Input Label passed is not a valid file, please re-check", this);
+    ShowErrorMessage("Input Atlas passed is not a valid file, please re-check", this);
     return;
   }
-  if (!cbica::isFile(inputatlas))
+  //read and store the entire data of csv file
+  std::vector< std::vector < std::string > > allRows; // store the entire data of the CSV file as a vector of columns and rows (vector< rows <cols> >)
+  std::string  inputFile = cbica::dos2unix(inputFileName, outputdirectory);
+  std::ifstream inFile(inputFile.c_str());
+  std::string csvPath = cbica::getFilenamePath(inputFile);
+  while (inFile.good())
   {
-    ShowErrorMessage("Input Atlas passed is not a valid file, please re-check", this);
+    std::string line;
+    std::getline(inFile, line);
+    line.erase(std::remove(line.begin(), line.end(), '"'), line.end());
+    if (!line.empty())
+    {
+      allRows.push_back(cbica::stringSplit(line, ","));
+    }
+  }
+  inFile.close();
+  std::cout << "input file parsed. Number of entries: " << allRows.size() << std::endl;
```

Comment:
cout statements may not be needed from GUI.

Reply:
removed